tests/test_sipApproximation.py

```diff
@@ -516,6 +1091,77 @@ def run(md, **kwds2):
         run(self.calexp03, order=3)
         run(self.wcs22, order=8)
 
+    def testCalculateSipWcsHeader(self):
+        """Test the calculateSipWcsHeader function
+
+        This function not only generates the coefficients approximating the WCS,
+        but allows creating a new ``SkyWcs`` using those coefficients.
+        """
+        def run(original, width, height, order, spacing=20, skyTol=1.0e-4, pixTol=1.0e-8):
+            """Run calculateSipWcsHeader and evaluate performance
+
+            Parameters
+            ----------
+            original : `lsst.afw.geom.SkyWcs`
+                Original WCS to approximate with SIP.
+            width, height : `int`
+                Dimensions of the image.
+            order : `int`
+                SIP order (equal to the maximum sum of the polynomial exponents).
+            spacing : `float`
+                Spacing between sample points.
+            skyTol : `float`
+                Tolerance in arcseconds for comparing sky positions.
+            pixTol : `float`
+                Tolerance in pixels for comparing pixel positions.
+
+            Returns
+            -------
+            wcs : `lsst.afw.geom.SkyWcs`
+                SIP WCS.
+            """
+            # Get new SIP WCS
+            bbox = Box2I(Point2I(0, 0), Extent2I(width, height))
+            header = calculateSipWcsHeader(original, order, bbox, spacing)
+            wcs = makeSkyWcs(header)
+
+            # Evaluate performance
+            points = [Point2D(xx, yy) for xx, yy in itertools.product(range(0, width, spacing),
+                                                                      range(0, height, spacing))]
+            coord1 = [original.pixelToSky(pp) for pp in points]
+            coord2 = [wcs.pixelToSky(pp) for pp in points]
+            offsets = (c1.getTangentPlaneOffset(c2) for c1, c2 in zip(coord1, coord2))
+            offsets = np.array([(off[0].asArcseconds(), off[1].asArcseconds()) for off in offsets])
+
+            self.assertFloatsAlmostEqual(offsets[0].mean(), 0.0, atol=skyTol)
+            self.assertFloatsAlmostEqual(offsets[1].mean(), 0.0, atol=skyTol)
+            self.assertFloatsAlmostEqual(offsets.mean(), 0.0, atol=skyTol)
+            self.assertFloatsAlmostEqual(offsets[0].std(), 0.0, atol=skyTol)
+            self.assertFloatsAlmostEqual(offsets[1].std(), 0.0, atol=skyTol)
+            self.assertFloatsAlmostEqual(offsets.std(), 0.0, atol=skyTol)
+            self.assertFloatsAlmostEqual(np.abs(offsets).max(), 0.0, atol=skyTol)
+
+            # Not comparing to absolute round-trip, but relative to the original WCS round-trip.
+            # This is an important distinction: we're recreating the original WCS, along with all its flaws.
```

Comment:
Apparently we're not checking this in Travis in afw, but I think it's best to break comments at 79 chars in new code.

Reply:
That's a bit annoying (the comment block starts at 15 chars in, which means the effective width is only 64 chars), but OK.